src/modules/DOM.js

```diff
@@ -16,16 +14,40 @@ document.getElementById('myonoffswitch').addEventListener('change',(event)=> {
 });
 
 document.getElementById('search-ic').addEventListener('click', (event)=> {
-
+    setTimeout(toggleLoadingModal,2000);
     const searchBar = document.getElementById('search-bar');
     searchWeatherByCity(searchBar.value.trim());
     searchBar.value = '';
-
+    toggleLoadingModal();
 });
+
 document.getElementById('search-bar').addEventListener('keyup', (e)=> {
```

Comment:
we can avoid all this by using a form to wrap the input field and  attach a submit event to it which will call a function to do our city search.